File: dart/luzid-sdk/lib/src/api/ping.dart
import 'package:luzid_grpc/luzid_grpc.dart';
import 'package:luzid_grpc_client/luzid_grpc_client.dart';

export 'package:luzid_grpc/luzid_grpc.dart' show AppOperation;

class LuzidPing {
  final PingClient _client;

  LuzidPing(LuzidGrpcClient client) : _client = client.ping;

  /// Performs a ping request to ensure the Luzid service is running.
  Future<PingResponse> ping() async {
    return _client.ping();
  }
}

File: ts/packages/luzid-sdk/src/api/ping.ts
import { PingResponse } from '@luzid/grpc'

import { LuzidGrpcClient } from '@luzid/grpc-client'

export { PingResponse }

export class LuzidPing {
  constructor(private readonly client: LuzidGrpcClient) {}

  /**
   * Pings the Luzid backend.
   */
  async ping(): Promise<PingResponse> {
    return this.client.ping.ping()
  }
}
